src/crypto/secretbox/error.rs

```diff
@@ -0,0 +1,25 @@
+//! Error types for the `secretbox` module
+
+/// A type representing error conditions of secret box operations
+#[derive(Debug, PartialEq)]
+pub enum SecretBoxError {
+    /// The ciphertext was too short. This results from using the `crypto_secretbox_open_easy`
+    /// function, which expects a message prepended with a MAC.
+    CiphertextTooShort,
+    /// The ciphertext failed verification.
+    CiphertextFailedVerification,
+}
+
+#[cfg(feature = "std")]
```

Comment:
Since `std::fmt::Display` is an alias for `core::fmt::Display`, it is possible to provide better user experience for no-std users by implementing the latter trait unconditionally, even though the `Error` trait could not be implemented with absence of std